SalesforcePluginApp.ts

```diff
@@ -0,0 +1,831 @@
+                          )
+                          .then((loginResponse) => {
+                            const loginResponseJSON = JSON.parse(
+                              loginResponse.content || '{}',
+                            );
+                            console.log(
+                              'Performing Salesforce Bot Login, Response:',
+                            );
+                            console.log(loginResponse);
+
+                            const setStatusHttpRequest: IHttpRequest = {
+                              headers: {
+                                'X-Auth-Token':
+                                  loginResponseJSON.data.authToken,
+                                'X-User-Id': loginResponseJSON.data.userId,
+                              },
+                              data: {
+                                message: 'online',
+                                status: 'online',
+                              },
+                            };
+
+                            http
+                              .post(
+                                `${rocketChatServerUrl}api/v1/users.setStatus`,
+                                setStatusHttpRequest,
+                              )
+                              .then(async (statusResponse) => {
+                                console.log(
+                                  'Setting Salesforce Bot Status, Response:',
+                                );
+                                console.log(statusResponse);
+
+                                const assoc = new RocketChatAssociationRecord(
+                                  RocketChatAssociationModel.ROOM,
+                                  message.room.id,
+                                );
+                                await persistence.createWithAssociation(
+                                  sessionIdParsedResponse,
+                                  assoc,
+                                );
+
+                                const roomId = message.room.id;
+                                const room: ILivechatRoom = (await read
+                                  .getRoomReader()
+                                  .getById(roomId)) as ILivechatRoom;
+                                // tslint:disable-next-line: max-line-length
+                                const targetDepartment: IDepartment = (await read
+                                  .getLivechatReader()
+                                  .getLivechatDepartmentByIdOrName(
+                                    targetDeptName,
+                                  )) as IDepartment;
+
+                                const transferData: ILivechatTransferData = {
+                                  currentRoom: room,
+                                  targetDepartment: targetDepartment.id,
+                                  targetAgent: targetagent,
+                                };
+
+                                await modify
```

Comment:
double checked and it looks like all my settings are right, but am still having issues with the handover... maybe we can add some sort of retry for some of these steps, or if there is any type of 'refresh status' api call that might help if the issue is due to the cached status not being updated or something

Reply:
Maybe the version im running of RC server is different than yours as well, could you point me at the checkout or fork you are using?

Reply:
I would suggest using the latest release of RC from here => https://github.com/RocketChat/Rocket.Chat/releases/tag/3.3.3

Reply:
Yes, the issue persists even on the latest release. I just checked. Looking into this error.

Reply:
I think I figured out the issue, not an error in the code actually just a config error from our side:

Just make sure the following Omnichannel `Available` button is ticked in our Salesforce agent user -
<img width="290" alt="Screenshot 2020-06-29 at 9 24 41 PM" src="https://user-images.githubusercontent.com/41849970/86028097-24c08180-ba4f-11ea-828b-2ed80ca3a8e0.png">

Just need to be ticked once after a new user is created. 

@David can you please check if this fixes the issue?

Reply:
Resolving this conversation in favour of this (commit)[https://github.com/RocketChat/Apps.Salesforce.LiveAgent/pull/1/commits/71dc10336144c8e12c544ac350a8679e7942ec4f].